src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.FruitDao;
+import core.basesyntax.db.FruitDaoImpl;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+    private final FruitDao fruitDao;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+        fruitDao = new FruitDaoImpl();
+    }
+
+    public FruitDao getFruitDao() {
+        return fruitDao;
```

Comment:
Exposing the `FruitDao` through a getter method violates the encapsulation principle. The `fruitDao` should be a private field used only within the `ShopServiceImpl` class. If you need to share the state of `fruitDao` with other classes, consider using a different approach that doesn't expose the internal implementation.